File: result/metagpt/claude/humaneval/108.c
#include <stdio.h>
#include <stdlib.h>
#include <math.h>

int count_nums(int* n, int size) {
    int num = 0;
    for (int i = 0; i < size; i++) {
        if (n[i] > 0) {
            num += 1;
        } else {
            int sum = 0;
            int w;
            w = abs(n[i]);
            while (w >= 10) {
                sum += w % 10;
                w = w / 10;
            }
            sum -= w;
            if (sum > 0) num += 1;
        }
    }
    return num;
}

File: result/metagpt/gpt4o/humaneval/108.rs
fn count_nums(n: &[i32]) -> i32 {
    let mut num = 0;
    for &value in n.iter() {
        if value > 0 {
            num += 1;
        } else {
            let mut sum = 0;
            let mut w = value.abs();
            while w >= 10 {
                sum += w % 10;
                w /= 10;
            }
            sum -= w;
            if sum > 0 {
                num += 1;
            }
        }
    }
    num
}

fn main() {
    let numbers = vec![12, -34, 56, -78, 90];
    let result = count_nums(&numbers);
    println!("Count of numbers meeting criteria: {}", result);
}